etcdserver/api/v3rpc/rpctypes/error.go

```diff
@@ -40,6 +40,8 @@ var (
 	ErrGRPCMemberNotEnoughStarted = status.New(codes.FailedPrecondition, "etcdserver: re-configuration failed due to not enough started members").Err()
 	ErrGRPCMemberBadURLs          = status.New(codes.InvalidArgument, "etcdserver: given member URLs are invalid").Err()
 	ErrGRPCMemberNotFound         = status.New(codes.NotFound, "etcdserver: member not found").Err()
+	ErrGRPCMemberNotLearner       = status.New(codes.FailedPrecondition, "etcdserver: can only promote a learner member").Err()
+	ErrGRPCLearnerNotReady        = status.New(codes.FailedPrecondition, "etcdserver: can only promote a learner member which catches up with leader").Err()
```

Comment:
is in sync with leader

Reply:
can we reuse the previously defined error message here? maybe define the error message as a const

Reply:
> can we reuse the previously defined error message here? maybe define the error message as a const

This is just following the exiting pattern. If we want to change it, it should be a separate PR for all existing errors.

Reply:
> is in sync with leader

sounds good.

Reply:
for this one, a const is better since we repeat the same message twice. but it might also apply to other error messages. i am not sure.